File: reni-frontend/src/main/reni-frontend/src/app/modules/intranet/services/ubigeo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { OutResponse } from '../../sesion/dto/response/out.response';
import { UbigeoDistritoListarRequest } from '../dto/request/ubigeo-distrito-listar.request';
import { UbigeoDistritoListarXUbigeoRequest } from '../dto/request/ubigeo-distrito-listar_x_ubig.request';
import { UbigeoProvinciaListarRequest } from '../dto/request/ubigeo-provincia-listar.request';
import { UbigeoProvinciaListarXUbigeoRequest } from '../dto/request/ubigeo-provincia-listar_x_ubig.request';
import { UbigeoDepartamentoListarResponse } from '../dto/response/ubigeo-departamento-listar.response';
import { UbigeoDistritoListarResponse } from '../dto/response/ubigeo-distrito-listar.response';
import { UbigeoProvinciaListarResponse } from '../dto/response/ubigeo-provincia-listar.response';

@Injectable()
export class UbigeoService {

  constructor(private http: HttpClient) { }

  public listarDepartamento(): Observable<OutResponse<UbigeoDepartamentoListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoDepartamentoListarResponse[]>>(`${environment.reniBackendUrl}/ubigeo/listarDepartamento`, {});
  }

  public listarProvincia(req: UbigeoProvinciaListarRequest): Observable<OutResponse<UbigeoProvinciaListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoProvinciaListarResponse[]>>(`${environment.reniBackendUrl}/ubigeo/listarProvincia`, req);
  }

  public listarDistrito(req: UbigeoDistritoListarRequest): Observable<OutResponse<UbigeoDistritoListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoDistritoListarResponse[]>>(`${environment.reniBackendUrl}/ubigeo/listarDistrito`, req);
  }

  public listarProvinciaXUbigeo(req: UbigeoProvinciaListarXUbigeoRequest): Observable<OutResponse<UbigeoProvinciaListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoProvinciaListarResponse[]>>(`${environment.reniBackendUrl}/ubigeo/listarProvinciaXUbigeo`, req);
  }

  public listarDistritoXUbigeo(req: UbigeoDistritoListarXUbigeoRequest): Observable<OutResponse<UbigeoDistritoListarResponse[]>> {
    return this.http.post<OutResponse<UbigeoDistritoListarResponse[]>>(`${environment.reniBackendUrl}/ubigeo/listarDistritoXUbigeo`, req);
  }
}

File: reni-backend/src/main/java/pe/gob/mimp/reni/service/UbigeoService.java
package pe.gob.mimp.reni.service;

import java.util.List;

import pe.gob.mimp.reni.dto.request.UbigeoDistritoListarRequest;
import pe.gob.mimp.reni.dto.request.UbigeoDistritoListarXUbigeoRequest;
import pe.gob.mimp.reni.dto.request.UbigeoProvinciaListarRequest;
import pe.gob.mimp.reni.dto.request.UbigeoProvinciaListarXUbigeoRequest;
import pe.gob.mimp.reni.dto.response.OutResponse;
import pe.gob.mimp.reni.dto.response.UbigeoDepartamentoListarResponse;
import pe.gob.mimp.reni.dto.response.UbigeoDistritoListarResponse;
import pe.gob.mimp.reni.dto.response.UbigeoObtenerDescUbigeoResponse;
import pe.gob.mimp.reni.dto.response.UbigeoProvinciaListarResponse;

public interface UbigeoService {

	public OutResponse<List<UbigeoDepartamentoListarResponse>> listarDepartamento();

	public OutResponse<List<UbigeoProvinciaListarResponse>> listarProvincia(UbigeoProvinciaListarRequest req);

	public OutResponse<List<UbigeoDistritoListarResponse>> listarDistrito(UbigeoDistritoListarRequest req);

	public OutResponse<List<UbigeoProvinciaListarResponse>> listarProvinciaXUbigeo(
			UbigeoProvinciaListarXUbigeoRequest req);

	public OutResponse<List<UbigeoDistritoListarResponse>> listarDistritoXUbigeo(
			UbigeoDistritoListarXUbigeoRequest req);

	public OutResponse<UbigeoObtenerDescUbigeoResponse> obtenerDescripcionUbigeo(String req);
}
